knowrob_common/prolog/owl.pl

```diff
@@ -593,6 +593,75 @@
 
 
 		 /*******************************
+		 *	  AUX PREDS (RD_FLIST)	*
+		 *******************************/
+
+%% Taken from semweb/rdf11. Needed to interpret PropertyChain axioms.
+
+kr_rdf_is_subject(T) :- atom(T).
+
+kr_rdf_list(RDFList, Prolog) :-
+    kr_rdf_is_subject(RDFList),
+    !,
+    kr_rdf_list_to_prolog(RDFList, Prolog).
+kr_rdf_list(RDFList, _Prolog) :-
+    type_error(rdf_subject, RDFList).
+
+kr_rdf_list_to_prolog(RDF, Prolog) :-
+    owl_same_as(RDF, 'http://www.w3.org/1999/02/22-rdf-syntax-ns#nil'),
+    !,
+    Prolog = [].
+kr_rdf_list_to_prolog(RDF, [H|T2]) :-
+    (   rdf_has(RDF, rdf:first, H0),
+        rdf_has(RDF, rdf:rest, T1)
+    *-> H = H0,
+        kr_rdf_list_to_prolog(T1, T2)
+    ;   type_error(rdf_list, RDF)
+    ).
+
+
+		 /*******************************
+		 *	  AUX PREDS (CIRC)	*
+		 *******************************/
+
+% An element in a property chain is either an object property or a description
+% containing an owl:inverseOf some object property.
```

Comment:
This check is needed because the RDFList returned when querying for PropertyChains may contain either object properties or 'anonymous' properties that are not individuals of owl:ObjectProperty, but do have an owl:inverseOf relation to an individual of type owl:'ObjectProperty'.